test/Etg.Yams.Core.Test/Process/EnvironmentTest.cs

```diff
@@ -0,0 +1,81 @@
+﻿using Etg.Yams.Utils;
+using System;
+using System.Linq;
+using System.Management.Automation;
+using System.Security;
+using Xunit;
+
+namespace Etg.Yams.Test.Process
+{
+    public class EnvironmentFixture : IDisposable
+    {
+        readonly string originalMachinePath;
+        public EnvironmentFixture()
+        {
+            originalMachinePath = Environment.GetEnvironmentVariable("PATH", EnvironmentVariableTarget.Machine);
+        }
+
+        public void Dispose()
+        {
+            try
+            {
+                Environment.SetEnvironmentVariable("PATH", originalMachinePath, EnvironmentVariableTarget.Machine);
+            }
+            catch (SecurityException e)
+            {
+                throw new Exception("Testing environment needs Admin permission to test changing the machine path -- it will be restored after the test!", e);
+            }
+        }
+    }
+
+    public class EnvironmentTest: IClassFixture<EnvironmentFixture>
+    {
+        [Theory]
+        [InlineData("C:\\Windows;C:\\Foo;", "C:\\Windows;D:\\Bar\\", "C:\\Windows;C:\\Foo;D:\\Bar\\;")]
+        [InlineData("C:\\Windows;C:\\Foo;", null, "C:\\Windows;C:\\Foo;")]
+        [InlineData(null, "C:\\Windows;C:\\Foo\\;", "C:\\Windows;C:\\Foo\\;")]
+        [InlineData("C:\\Windows;C:\\Foo", "C:\\Windows;C:\\Foo\\;", "C:\\Windows;C:\\Foo;C:\\Foo\\;")]
+        public void TestMergePath(string processPath, string machinePath, string expectedResult)
+        {
+            string actualResult = EnvironmentUtils.MergePath(processPath, machinePath);
+            Assert.Equal(actualResult, expectedResult);
+        }
+
+        [Fact]
+        public void TestEnvironmentExpansion()
+        {
+            // Setup: Add an entry to the machine path from another process
+            using (PowerShell shell = PowerShell.Create())
```

Comment:
I'd rather not include this test or the requirement to run visual studio in admin mode to run tests. I see the value of this tests as it verifies the functionality end to end but sometime it's a tradeoff between the maintainability of tests and how much value does that test add. I am pretty happy with the test you added above.